ingestion/src/metadata/ingestion/source/pipeline/openlineage/metadata.py

```diff
@@ -0,0 +1,534 @@
+        run facet so we can always create pipeline entities and link them to lineage events.
+
+        :param run_facet: Open Lineage run facet
+        :return: pipeline name (not fully qualified name)
+        """
+        run_facet = pipeline_details.run_facet
+
+        namespace = run_facet["facets"]["parent"]["job"]["namespace"]
+        name = run_facet["facets"]["parent"]["job"]["name"]
+
+        return f"{namespace}-{name}"
+
+    @classmethod
+    def _filter_event_by_type(
+        cls, event: OpenLineageEvent, event_type: EventType
+    ) -> Optional[Dict]:
+        """
+        returns event if it's of particular event_type.
+        for example - for lineage events we will be only looking for EventType.COMPLETE event type.
+
+        :param event: Open Lineage raw event.
+        :param event_type: type of event we are looking for.
+        :return: Open Lineage event if matches event_type, otherwise None
+        """
+        return event if event.event_type == event_type else {}
+
+    @classmethod
+    def _get_om_table_columns(cls, table_input: Dict) -> Optional[List]:
+        """
+
+        :param table_input:
+        :return:
+        """
+        try:
+            fields = table_input["facets"]["schema"]["fields"]
+
+            # @todo check if this way of passing type is ok
+            columns = [
+                Column(name=f.get("name"), dataType=f.get("type").upper())
+                for f in fields
+            ]
+            return columns
+        except KeyError:
+            return None
+
+    def get_create_table_request(self, table: Dict) -> Optional[Either]:
+        """
+        If certain table from Open Lineage events doesn't already exist in Open Metadata, register appropriate entity.
+        This makes sense especially for output facet of OpenLineage event - as database service ingestion is a scheduled
+        process we might fall into situation where we received Open Lineage event about creation of a table that is yet
+        to be ingested by database service ingestion process. To avoid missing on such lineage scenarios, we will create
+        table entity beforehand.
+
+        :param table: single object from inputs/outputs facet
+        :return: request to create the entity (if needed)
+        """
+        om_table_fqn = None
+
+        try:
+            table_details = OpenlineageSource._get_table_details(table)
+        except ValueError as e:
+            return Either(
+                left=StackTraceError(
+                    name="",
+                    error=f"Failed to get partial table name: {e}",
+                    stackTrace=traceback.format_exc(),
+                )
+            )
+        try:
+            om_table_fqn = self._get_table_fqn_from_om(table_details)
+
+            # if fqn found then it means table is already registered and we don't need to render create table request
+            return None
+        except FQNNotFoundException:
+            pass
```

Comment:
instead of doing a pass we can add some debug logs

Reply:
bump